integration/entity.go

```diff
@@ -102,15 +102,18 @@ func (e *Entity) isLocalEntity() bool {
 
 // SameAs return true when is same entity
 func (e *Entity) SameAs(b *Entity) bool {
+	b.lock.Lock()
+	defer b.lock.Unlock()
 	if e.Metadata == nil || b.Metadata == nil {
```

Comment:
out of curiosity, as `e.Metadata` is read here, shouldn't be also locked? Maybe I'm not totally getting the case but it  seems weird to lock `b` and not `e`

Reply:
You are right, this responsibility shouldn't be delegated to the e entity. I've done it properly in the last commit, I've added a Mutex to the entityMetadata struct, that at the end is the one creating the race condition and I've locked the Key endpoint, and now everything works the same way but with a better design, thanks for pointing this out ;)